### PR TITLE
Improvment/backend/suggest api

### PR DESCRIPTION
**Pull request by sapiens-sapide**
**NB** :  
To benefit from the new suggest and search features, elasticsearch mappings need to be changed.
A new cli command — `caliopen migrate_index` — comes with this PR, as well as a script to migrate indexes from v0 to v2.
**NB** :  
The migration from v0 to v2 will create aliases for users, to ease futur migrations. This time, and this time only, ES service should be unavailable for users during the migration to prevent data corruption.

Example to run this first migration : `caliopen -f caliopen.yaml.template migrate_index -s index_migration_v0_to_v2.py`

### CODE REVIEW

File: src/backend/main/py.main/caliopen_main/user/core/new_user_setups.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
```
Comment (gdchamal) [inline src/backend/main/py.main/caliopen_main/user/core/new_user_setups.py]: name to setups.py not a new_user_setups.py

File: src/backend/main/py.main/caliopen_main/user/core/user.py [hunk 2]
```diff
@@ -185,12 +185,13 @@ def create(cls, new_user):
...
         #      create user and linked contact
         """
+
         def rollback_username_storage(username):
             UserName.get(username).delete()
 
         # 0. check for user email white list and max number of users
-        cls._check_whitelistes(new_user)
-        cls._check_max_users()
+        # cls._check_whitelistes(new_user)
```
Comment (gdchamal) [inline src/backend/main/py.main/caliopen_main/user/core/user.py]: uncomment that 2 lines that must still active

File: src/backend/main/py.storage/caliopen_storage/store/model.py [hunk 3]
```diff
@@ -69,9 +66,21 @@ def client(cls):
...
+    def create_mapping(cls, user_id):
+        """Create and save elasticsearch mapping for the cls.doc_type."""
+
+        if hasattr(cls, 'build_mapping'):
+            log.info('Create index {} mapping for doc_type {}'.
+                     format(user_id, cls.doc_type))
+            try:
+                cls.build_mapping().save(using=cls.client(), index=user_id)
+            except Exception as exc:
+                log.warn(
```
Comment (gdchamal) [inline src/backend/main/py.storage/caliopen_storage/store/model.py]: error not warn

File: src/backend/tools/py.CLI/caliopen_cli/commands/migrate_index.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+def load_from_file(filepath):
+    c = None
+    expected_class = 'IndexMigrator'
+
+    mod_name, file_ext = os.path.splitext(os.path.split(filepath)[-1])
+
+    if file_ext.lower() == '.py':
+        py_mod = imp.load_source(mod_name, filepath)
+
+    elif file_ext.lower() == '.pyc':
```
Comment (gdchamal) [inline src/backend/tools/py.CLI/caliopen_cli/commands/migrate_index.py]: load only if .py file, not .pyc

File: src/backend/main/py.main/caliopen_main/user/core/new_user_setups.py (hunk 1 above)
Comment (sapiens-sapide) [reply src/backend/main/py.main/caliopen_main/user/core/new_user_setups.py]: => commit #68c73ee

File: src/backend/main/py.main/caliopen_main/user/core/user.py (hunk 2 above)
Comment (sapiens-sapide) [reply src/backend/main/py.main/caliopen_main/user/core/user.py]: => commit #68c73ee

File: src/backend/main/py.storage/caliopen_storage/store/model.py (hunk 3 above)
Comment (sapiens-sapide) [reply src/backend/main/py.storage/caliopen_storage/store/model.py]: => commit #236ad67

File: src/backend/tools/py.CLI/caliopen_cli/commands/migrate_index.py (hunk 4 above)
Comment (sapiens-sapide) [reply src/backend/tools/py.CLI/caliopen_cli/commands/migrate_index.py]: => commit #0729736

**Pull request merged**